### ISSUE TITLE
block-scoped-var gives a false positive on function argument destructuring

### ISSUE THREAD

**Issue by vasa-chi** [eslint/eslint#1996]
```shell
$ eslint --version
v0.16.1
```

.eslintrc

```json
{
  "env": {
    "es6": true
  },
  "rules": {
    "block-scoped-var": 2
  },
  "ecmaFeatures": {
    "destructuring": true
  }
}
```

test.js
```js
'use strict';

var fun = function({x}) {
  return x;
};

var fun2 = function(arg) {
  var {x} = arg;
  return x;
};

fun({x: 2});
fun2({x: 2});
```

Error
```
test.js
  4:9  error  "x" used outside of binding context  block-scoped-var

✖ 1 problem (1 error, 0 warnings)
```

As you can see, the second function is linted correctly (due to #1863), but the first one fails.

**Comment by nzakas** [comment]
Yeah, we are probably just not dealing with destructured args properly.

**Comment by nzakas** [comment]
Working on this.
